lib/composer.js

```diff
@@ -194,6 +194,10 @@ export default class Composer extends Disposable {
 
     await Promise.all(jobs)
 
+    if (this.shouldCleanAfterBuild()) {
```

Comment:
@thomasjo We could also have `buildJob` return false if an error is thrown or if an error is sent to the log:

```javascript
  async buildJob (builder, jobState) {
    try {
      const statusCode = await builder.run(jobState)
      builder.parseLogAndFdbFiles(jobState)

      const messages = jobState.getLogMessages() || []
      for (const message of messages) {
        latex.log.showMessage(message)
      }

      if (statusCode > 0 || !jobState.getLogMessages() || !jobState.getOutputFilePath()) {
        this.showError(jobState)
        return false
      }

      if (this.shouldMoveResult(jobState)) {
        this.moveResult(jobState)
      }
      this.showResult(jobState)

      return messages.every(message => message.type !== 'error')
    } catch (error) {
      latex.log.error(error.message)
    }

    return false
  }
```

Throughts?